android/src/main/java/com/reactlibrary/RNPurchasesModule.java

```diff
@@ -102,10 +102,24 @@ private WritableMap mapForSkuDetails(final SkuDetails detail) {
         }
         map.putString("intro_price_string", detail.getIntroductoryPrice());
         map.putString("intro_price_period", detail.getIntroductoryPricePeriod());
+        if (detail.getIntroductoryPricePeriod() != null && !detail.getIntroductoryPricePeriod().isEmpty()) {
+            RNPurchasesPeriod period = RNPurchasesPeriod.parse(detail.getIntroductoryPricePeriod());
+            if (period.years > 0) {
+                map.putString("intro_price_period_unit", "YEAR");
+                map.putInt("intro_price_period_number_of_units", period.years);
+            } else if (period.months > 0) {
+                map.putString("intro_price_period_unit", "MONTH");
+                map.putInt("intro_price_period_number_of_units", period.months);
+            } else if (period.days > 0) {
+                map.putString("intro_price_period_unit", "DAY");
+                map.putInt("intro_price_period_number_of_units", period.days);
+            }
+        } else {
+            map.putString("intro_price_period_unit", "");
```

Comment:
Why not null for these,?

Reply:
I did like that to mimic `intro_price_period`. `map.putString("intro_price_period", detail.getIntroductoryPricePeriod());` would send to JavaScript`intro_price_period: ""` if there is no introductory price. So it's just to keep it consistent. I agree the JavaScript object shouldn't contain "" but it has been like that forever and changing it would break implementations.

Reply:
Sounds good, also per Slack discussion upstream vendors are also using empty strings (gross 🤢 )